frontend/src/scenes/feature-flags/FeatureFlagReleaseConditions.tsx

```diff
@@ -24,14 +24,67 @@ import { urls } from 'scenes/urls'
 
 import { cohortsModel } from '~/models/cohortsModel'
 import { groupsModel } from '~/models/groupsModel'
-import { AnyPropertyFilter, FeatureFlagGroupType } from '~/types'
+import { AnyPropertyFilter, FeatureFlagGroupType, PropertyOperator } from '~/types'
 
 import { featureFlagLogic } from './featureFlagLogic'
 import {
     featureFlagReleaseConditionsLogic,
     FeatureFlagReleaseConditionsLogicProps,
 } from './FeatureFlagReleaseConditionsLogic'
 
+function PropertyValueComponent({
+    property,
+    cohortsById,
+}: {
+    property: AnyPropertyFilter
+    cohortsById: Record<string, any>
+}): JSX.Element {
+    if (property.type === 'cohort') {
+        return (
+            <LemonButton
+                type="secondary"
+                size="xsmall"
+                to={urls.cohort(property.value)}
+                sideIcon={<IconOpenInNew />}
+                targetBlank
+            >
+                {(property.value && cohortsById[property.value]?.name) || `ID ${property.value}`}
+            </LemonButton>
+        )
+    }
+
+    if (property.value === PropertyOperator.IsNotSet || property.value === PropertyOperator.IsSet) {
+        return <></>
+    }
```

Comment:
only logic changes on the PR, otherwise I just moved the lines into a new function for easier readability